src/main/java/org/folio/linked/data/mapper/kafka/inventory/InventoryInstanceMapper.java

```diff
@@ -0,0 +1,18 @@
+package org.folio.linked.data.mapper.kafka.inventory;
+
+import static org.folio.linked.data.util.Constants.SEARCH_WORK_RESOURCE_NAME;
+import static org.mapstruct.MappingConstants.ComponentModel.SPRING;
+
+import java.util.UUID;
+import org.folio.linked.data.domain.dto.InventoryInstanceEvent;
+import org.folio.linked.data.domain.dto.ResourceIndexEvent;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(componentModel = SPRING, imports = UUID.class)
+public abstract class InventoryInstanceMapper {
+
+  @Mapping(target = "id", expression = "java(UUID.randomUUID().toString())")
+  @Mapping(target = "resourceName", constant = SEARCH_WORK_RESOURCE_NAME)
+  public abstract ResourceIndexEvent toReindexEvent(InventoryInstanceEvent inventoryInstanceEvent);
```

Comment:
Such event will contain only ID and the suppress flags. Could we check if mod-search updates indexed data as we expected here?

Reply:
Sure, will double check this. ResourceIndexEvent will contain id, resourceName, type and entire _new and old objects as per the mapping:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/9ca110cb-6046-47fd-9488-f1b8c91eeaf9">

In mod-search we have ResourceEvent with same structure:
<img width="414" alt="image" src="https://github.com/user-attachments/assets/0d4de796-6425-4d28-98ce-ac907cd86408">